### ISSUE TITLE
interface for `stop_iter` validation set

### ISSUE THREAD

**Issue by simonpcouch** [tidymodels/bonsai#17]
`stop_iter` can be used to set how many iterations without an improvement in the objective function occur before training should be halted.

xgboost allows for passing a proportion of the training set to use as an internal validation set via `parameters` (`?details_boost_tree_xgboost`).

When `stop_iter` is supplied (as `early_stopping_rounds`), a `valids` argument is required by lightgbm: `”a list of lgb.Dataset objects, used for validation”`. This seems like a much more cumbersome interface than a proportion—a good portion of bonsai's lightgbm infrastructure is taking care of the formatting for `lgb.Dataset` objects. Allowing for passing a proportion of the training data feels much more straightforward, but a possibly confusing interface given that other `set_engine` arguments are always in the format that the modeling function takes them. As of now, the training set is passed.
